lib/puppet/provider/reboot_notify/notify.rb

```diff
@@ -20,11 +20,15 @@ def exists?
   end
 
   def create
-    File.open(@target,'w'){|fh| fh.puts(PSON.pretty_generate({}))}
+    begin
+      File.open(@target,'w'){|fh| fh.puts(JSON.pretty_generate({}))}
+    rescue
```

Comment:
I'm not a fan of blanket rescues and would prefer to see rescue of specific error types or their base class, and a reporting of the error they provide.  For example,

rescue Errno::ENOENT, Errno::EACCES, JSON::JSONError, ArgumentError, TypeError => e
  *OR*
rescue SystemCallError, JSON::JSONError => e
  raise(Puppet::Error, "reboot_notify:  Could not create '#{@target}' - #{e.message}")